rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationHandler.java

```diff
@@ -129,7 +128,6 @@ public Integration create(@ConvertGroup(from = Default.class, to = AllValidation
             .createFrom(integration)
             .deployedRevisionId(revision.getVersion())
             .addRevision(revision)
-            .userId(Tokens.getUsername())
```

Comment:
This is currently an issue as the check for how many integrations are allowed is connected to the userName which is put as a label to the resource objects (see. e.g. [here](https://github.com/rhuss/ipaas-rest/blob/41b954a25817498ab68cb4351013ffc7d0b56bfe/controllers/src/main/java/io/syndesis/controllers/integration/online/ActivateHandler.java#L115)). 

Despite the fact, I that the limitation should be global anyway (not per user), so this can be changed easily, I guess at least when we reach to the "User & Teams" use case we need a way to get to the subject of the current  http request (which might need to be added to the various domain objects for auditing purposes).

Is it easy to extract such a subject from Spring Security ? Maybe we can already add it here as user id ?

Reply:
Should be super easy you can either use the [static helpers](https://docs.spring.io/spring-security/site/docs/current/apidocs/org/springframework/security/core/context/SecurityContextHolder.html#getContext--)

    SecurityContextHolder.getContext().getAuthentication()

Or you can inject it into a REST handler method I think there should be one for JAX-RS, in spring-web it's super simple, just add Authentication parameter to the method.

Reply:
@jimmidyson could change this line to use the Spring Security subject as the userId here ? As said, this is required currently for the integration limit feature.

Reply:
Updated